curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/review-dom-manipulation-by-building-a-rock-paper-scissors-game/663d2f8056542a8066fd6cc6.md

```diff
@@ -0,0 +1,233 @@
+---
+id: 663d2f8056542a8066fd6cc6
+title: Step 3
+challengeType: 0
+dashedName: step-3
+---
+
+# --description--
+
+Now it is time to get the results of the round. Complete the `getRoundResults` function. 
+
+If the player wins the round, update the `playerScore` by `1` and return the message `"Player wins! [player's choice] beats [computer's choice]"`.
+
+If the computer and player choose the same option, return the message `"It's a tie! Both chose [player's choice]"`.
+
+If the computer wins the round, update the `computerScore` by `1` and return the message `"Computer wins! [computer's choice] beats [player's choice]"`.
+
+`[computer's choice]` should be replaced with `computerResult` while `[player's choice]` should be replaced with the `userOption`.
+
+**Tips**
+
+- Remember you can use the `hasPlayerWonTheRound` function to check if the player wins the round.
+- You can use template literals or regular string concatenation to build the message.
+
+
+# --hints--
+
+Your `getRoundResults` should return a string.
+
+```js
+assert.equal(typeof getRoundResults("Rock"), "string")
+```
+
+Your `getRoundResults` function should return the correct message based on who wins the round. If no one wins, the message should say it's a tie.
+
+```js
+const result = getRoundResults("Paper")
+if(result.startsWith("Player wins!")){
+  assert.strictEqual(result, "Player wins! Paper beats Rock")
+  assert.strictEqual(playerScore, 1)
+} else if(result.startsWith("Computer wins!")) {
+  assert.strictEqual(result, "Computer wins! Scissors beats Paper")
+  assert.strictEqual(computerScore, 1)
+} else if(result.startsWith("It's a tie!")){
+  assert.strictEqual(result, "It's a tie! Both chose Paper")
+}
+```
```

Comment:
````suggestion
```js
const result = getRoundResults("Paper")
if(result.startsWith("Player wins!")){
  assert.strictEqual(result, "Player wins! Paper beats Rock")
  assert.isAtLeast(playerScore, 1)
} else if(result.startsWith("Computer wins!")) {
  assert.strictEqual(result, "Computer wins! Scissors beats Paper")
  assert.isAtLeast(computerScore, 1)
} else if(result.startsWith("It's a tie!")){
  assert.strictEqual(result, "It's a tie! Both chose Paper")
}
```
````

Okay, the issue with the tests is because the previous test in this step runs the `getRoundResults` function to confirm the return type is a string.


Which means it is possible for the player to win twice in a row, resulting in a score of `2` instead of `1`.

Reply:
Alternatively, we could leave this as is and reset both of the variables to 0 before this test.

Reply:
ahhh ok cool. That makes sense. It looks like that might be an issue in other steps, so I will update those too. Thanks